src/FFmpegWriter.cpp

```diff
@@ -1904,6 +1907,10 @@ void FFmpegWriter::write_audio_packets(bool is_final) {
             ZmqLogger::Instance()->AppendDebugMethod("FFmpegWriter::write_audio_packets ERROR [" + (std::string) av_err2str(error_code) + "]", "error_code", error_code);
         }
 
+        // Increment PTS (in samples)
+        write_audio_count += FFMIN(audio_input_frame_size, audio_input_position);
+        frame_final->pts = write_audio_count; // Set the AVFrame's PTS
```

Comment:
This feels like the same error, just moved to a different place. Wouldn't we want to increment _after_ setting the timestamp for the current frame, not before? That'll then prepare `write_audio_count` (that is a **terrible** variable name, for the record) to become the timestamp of the _next_ frame.

```suggestion
        // Increment PTS (in samples)
        frame_final->pts = write_audio_count; // Set the AVFrame's PTS
        write_audio_count += FFMIN(audio_input_frame_size, audio_input_position);
```

Reply:
Fixing this